circuits/plonk/src/constraints.rs

```diff
@@ -84,22 +84,23 @@ pub struct ConstraintSystem<F: FftField> {
     pub fr_sponge_params: ArithmeticSpongeParams<F>,
 }
 
-pub fn zk_w<F: FftField>(domain: D<F>) -> F {
-    domain.group_gen.pow(&[domain.size - 3])
+pub fn zk_w<F: FftField>(domain: D<F>) -> Option<F> {
+    if domain.size < 3 { return None }
+    Some(domain.group_gen.pow(&[domain.size - 3]))
```

Comment:
I think it's cleaner to use `checked_sub` for overflows/underflows as you don't need to think about the edge case:

```rust
domain.size.checked_sub(3).map(|idx| 
  domain.group_gen.pow(&[idx])
);
```

Reply:
this seems better to me also